### ISSUE TITLE
Bug: `indent`  `offsetTernaryExpressions` has issue with `function`

### ISSUE THREAD

**Issue by loongzhu** [eslint/eslint#19156]
### Environment

Node version: v20.18.0
npm version: v10.8.2
Local ESLint version: v9.13.0 (Currently used)
Global ESLint version: Not found
Operating System: win32 10.0.22631

### What parser are you using?

@typescript-eslint/parser

### What did you do?

<details>
<summary>Configuration</summary>

```ts
import pluginJs from "@eslint/js";
import globals from "globals";
import tseslint from "typescript-eslint";

export default [
	{ languageOptions: { globals: globals.browser } },
	pluginJs.configs.recommended,
	...tseslint.configs.recommended,

	{
		files: ["**/*.{js,mjs,cjs,ts}"],
		rules: {
			// indent: ["error", "tab", { offsetTernaryExpressions: false }],
			indent: ["error", "tab", { offsetTernaryExpressions: true }],
			"no-console": ["error", { allow: ["warn", "error"] }],
			"no-unused-vars": "off",
			"@typescript-eslint/no-unused-vars": [
				"error",
				{ argsIgnorePattern: "^_", varsIgnorePattern: "^_" },
			],
			"no-redeclare": "off",
			"@typescript-eslint/no-redeclare": "error",
			"@typescript-eslint/no-unused-expressions": [
				"error",
				{ allowShortCircuit: true, allowTernary: true },
			],
		},
	},
	{
		ignores: ["/build/*", ".config/*", "dist/*", "out/*"],
	},
];

```
</details>

```ts
const list: string[] = [];

const t = (string: string, _data: { num: number }) => string;

const _obj = {
	condition:
		list.length > 3
			? t("string", {
					num: list.length,
				})
			: "",
};

const _func = () => {
	return list.length > 3
		? () => {
				return true;
			}
		: "";
};

```

### What did you expect to happen?

no error

![Image](https://github.com/user-attachments/assets/eeaccdb3-aefa-41f1-ae79-d8fa9d9e3d8e)

the same in eslint play

![Image](https://github.com/user-attachments/assets/07aa7731-11b3-43e3-842a-4c0f56c2d36f)

### What actually happened?

```bash
D:\WorkSpace\ts-demo\src\lint.ts
   9:1  error  Expected indentation of 4 tabs but found 5  indent
  10:1  error  Expected indentation of 3 tabs but found 4  indent
```

after run `pnpm eslint --fix`, it becomes  this

![Image](https://github.com/user-attachments/assets/cc03bc7a-25e1-4b4c-9aef-4301e9cb85c1)

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgaW5kZW50OiBbXCJlcnJvclwiLCAyLCB7IFwib2Zmc2V0VGVybmFyeUV4cHJlc3Npb25zXCI6IHRydWUgfV0qL1xuXG5mdW5jdGlvbiB0KHN0cixkYXRhKSB7fVxuXG5jb25kaXRpb25cbiAgPyAoKSA9PiB7XG4gICAgICByZXR1cm5cbiAgICB9XG4gIDogYVxuXG5jb25kaXRpb25cbiAgPyB0KCdzdHJpbmcnLHtcbiAgICAgIGRhdGE6MVxuICAgIH0pXG4gIDogYVxuIiwib3B0aW9ucyI6eyJydWxlcyI6e30sImxhbmd1YWdlT3B0aW9ucyI6eyJzb3VyY2VUeXBlIjoibW9kdWxlIiwicGFyc2VyT3B0aW9ucyI6eyJlY21hRmVhdHVyZXMiOnt9fX19fQ==

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by fasttime** [comment]
Hi @loongzhu, note that the [`indent`](https://eslint.org/docs/latest/rules/indent) rule is deprecated: we no longer maintain it, and we don't recommend using it. As stated in the documentation, you could switch to the [corresponding rule](https://eslint.style/rules/js/indent) in [`@stylistic/eslint-plugin-js`](https://eslint.style/packages/js).

**Comment by loongzhu** [comment]
ok thanks, I just noticed the deprecation notice
